### PR TITLE
kubernetes: more lazy use of TextEncoder

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Yet another fix for #19925 🎉 😁 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by freben — APPROVED** [review]
Yep, I like it. Maybe we should even document this pattern somewhere in a faq or adr or something. 

Can we also remove the setupTests stuff in `app` as part of this?

File: plugins/kubernetes/src/components/PodExecTerminal/PodExecTerminalAttachAddon.ts [hunk 1]
```diff
@@ -15,9 +15,9 @@
  */
 import { AttachAddon, IAttachOptions } from 'xterm-addon-attach';
 
-const textEncoder = new TextEncoder();
-
 export class PodExecTerminalAttachAddon extends AttachAddon {
+  #textEncoder = new TextEncoder();
```
Comment (luchillo17) [inline plugins/kubernetes/src/components/PodExecTerminal/PodExecTerminalAttachAddon.ts:19]: What's this `#` convention? seen `$` to mean private variables or observables but can't remember seeing this one...

File: plugins/kubernetes/src/components/PodExecTerminal/PodExecTerminalAttachAddon.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ export class PodExecTerminalAttachAddon extends AttachAddon {
         return;
       }
 
-      const buffer = Uint8Array.from([0, ...textEncoder.encode(data)]);
+      const buffer = Uint8Array.from([0, ...this.#textEncoder.encode(data)]);
```
Comment (luchillo17) [inline plugins/kubernetes/src/components/PodExecTerminal/PodExecTerminalAttachAddon.ts:33]: What's the need for a local variable in the class? What difference does it make compared to the scoped constant at the top of the file?

File: plugins/kubernetes/src/components/PodExecTerminal/PodExecTerminalAttachAddon.ts (hunk 1 above)
Comment (freben) [reply plugins/kubernetes/src/components/PodExecTerminal/PodExecTerminalAttachAddon.ts:19]: This is the _native_ javascript syntax! https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

File: plugins/kubernetes/src/components/PodExecTerminal/PodExecTerminalAttachAddon.ts (hunk 2 above)
Comment (freben) [reply plugins/kubernetes/src/components/PodExecTerminal/PodExecTerminalAttachAddon.ts:33]: Having it at the root of the file makes it get executed _instantly_ when the module is imported, no matter if the class is used or not. Having it inside the class, means that the `TextEncoder` only gets instantiated when the class gets instantiated (which does NOT happen inside `packages/app` test runs).

Comment (luchillo17) [reply plugins/kubernetes/src/components/PodExecTerminal/PodExecTerminalAttachAddon.ts:33]: That's actually a big difference, thanks for clarifying.

**Review by luchillo17 — APPROVED** [review]

**Review by mclarke47 — APPROVED** [review]

**Pull request merged**